automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/ClusterMonitoringSpec.scala

```diff
@@ -268,6 +272,71 @@ class ClusterMonitoringSpec extends FreeSpec with LeonardoTestUtils with Paralle
         }
       }
     }
+
+    "should download file as pdf" in {
+      withProject { project => implicit token =>
+        withNewGoogleBucket(project) { bucketName =>
+
+          val ronPetServiceAccount = Sam.user.petServiceAccountEmail(project.value)(ronAuthToken)
+          googleStorageDAO.setBucketAccessControl(bucketName, EmailGcsEntity(GcsEntityTypes.User, ronPetServiceAccount), GcsRoles.Owner)
+
+          val userScriptString = "#!/usr/bin/env bash\n\npip install nbconvert\napt-get install -yq pandoc texlive-xetex"
+          val userScriptObjectName = GcsObjectName("user-script-downloadaspdf.sh")
+          val userScriptUri = s"gs://${bucketName.value}/${userScriptObjectName.value}"
+
+          withNewBucketObject(bucketName, userScriptObjectName, userScriptString, "application/pdf") { objectName =>
+            googleStorageDAO.setObjectAccessControl(bucketName, objectName, EmailGcsEntity(GcsEntityTypes.User, ronPetServiceAccount), GcsRoles.Owner)
+            val clusterName = ClusterName("user-script-cluster" + makeRandomId())
+
+            withNewCluster(project, clusterName, ClusterRequest(Map(), None, Option(userScriptUri)), monitorDelete = false) { cluster =>
+              val download = makeTempDownloadDirectory()
+              withWebDriver(download) { implicit driver =>
+                withNewNotebook(cluster) { notebookPage =>
+                  notebookPage.executeCell("1+1") shouldBe Some("2")
+                  notebookPage.downloadAsPdf()
+                  val notebookName = notebookPage.currentUrl.substring(notebookPage.currentUrl.lastIndexOf('/') + 1, notebookPage.currentUrl.lastIndexOf('?')).replace(".ipynb", ".pdf")
+                  // sanity check the file downloaded correctly
+                  val downloadFile = new File(download, notebookName)
+                  downloadFile.deleteOnExit()
+                  logger.info(s"download: $downloadFile")
+                  implicit val patienceConfig: PatienceConfig = getAfterCreatePatience
+                  eventually {
+                    assert(downloadFile.exists(), s"Timed out (${patienceConfig.timeout} seconds) waiting for file.exists $downloadFile")
+                    assert(downloadFile.isFile(), s"Timed out (${patienceConfig.timeout} seconds) waiting for file.isFile $downloadFile")
+                  }
+                }
+              }
+            }(ronAuthToken)
+          }
+        }
+      }
+    }
+  }
+
+  private def makeTempDownloadDirectory(): String = {
```

Comment:
Both UI and Leo Selenium tests require a temporary download directory. I created PR to move this common method to workbench-lib `WebBrowserSpec`. Once PR is merged, this method can be called from `WebBrowserSpec`

https://github.com/broadinstitute/workbench-libs/pull/152